src/libgit2/repository.c

```diff
@@ -4009,3 +4009,28 @@ int git_repository_commit_parents(git_commitarray *out, git_repository *repo)
 	git_reference_free(head_ref);
 	return error;
 }
+
+int git_repository__abbrev_length(int *out, git_repository *repo)
+{
+	size_t oid_hexsize;
+	int len;
+	int error;
+
+	oid_hexsize = git_oid_hexsize(repo->oid_type);
+
+	if ((error = git_repository__configmap_lookup(&len, repo, GIT_CONFIGMAP_ABBREV)) < 0)
+		return error;
+
+	if (len == GIT_ABBREV_FALSE) {
+		len = (int)oid_hexsize;
+	}
+
+	if (len < GIT_ABBREV_MINIMUM || (size_t)len > oid_hexsize) {
+		git_error_set(GIT_ERROR_CONFIG, "invalid oid abbreviation setting: '%d'", len);
+		return -1;
+	}
```

Comment:
I think that we should invert these tests so that we can handle large values.

```suggestion
	if (len < GIT_ABBREV_MINIMUM) {
		git_error_set(GIT_ERROR_CONFIG, "invalid oid abbreviation setting: '%d'", len);
		return -1;
	}

	if (len == GIT_ABBREV_FALSE || (size_t)len > oid_hexsize)
		len = (int)oid_hexsize;
```